cluster-autoscaler/core/scale_up.go

```diff
@@ -632,6 +632,75 @@ func ScaleUp(context *context.AutoscalingContext, processors *ca_processors.Auto
 	}, nil
 }
 
+// ScaleUpToNodeGroupMinSize tries to scale up node groups that have less nodes than the configured min size.
+// The source of truth for the current node group size is the TargetSize queried directly from cloud providers.
+// Return the scale up status (ScaleUpNotNeeded, ScaleUpSuccessful or FailedResizeNodeGroups) and errors if any.
+func ScaleUpToNodeGroupMinSize(context *context.AutoscalingContext, clusterStateRegistry *clusterstate.ClusterStateRegistry,
+	nodeInfos map[string]*schedulerframework.NodeInfo) (*status.ScaleUpStatus, errors.AutoscalerError) {
+	now := time.Now()
+	nodeGroups := context.CloudProvider.NodeGroups()
+	gpuLabel := context.CloudProvider.GPULabel()
+	availableGPUTypes := context.CloudProvider.GetAvailableGPUTypes()
+	scaleUpInfos := make([]nodegroupset.ScaleUpInfo, 0)
+
+	for _, ng := range nodeGroups {
+		if !ng.Exist() {
+			klog.Warningf("ScaleUpToNodeGroupMinSize: NodeGroup %s does not exist", ng.Id())
+			continue
+		}
+
+		targetSize, err := ng.TargetSize()
+		if err != nil {
+			klog.Warningf("ScaleUpToNodeGroupMinSize: failed to get target size of node group %s", ng.Id())
+			continue
+		}
+
+		klog.V(4).Infof("ScaleUpToNodeGroupMinSize: NodeGroup %s, TargetSize %d, MinSize %d, MaxSize %d", ng.Id(), targetSize, ng.MinSize(), ng.MaxSize())
+		if targetSize >= ng.MinSize() {
+			continue
+		}
+
+		info := nodegroupset.ScaleUpInfo{
+			Group:       ng,
+			CurrentSize: targetSize,
+			NewSize:     ng.MinSize(),
```

Comment:
This should respect resource limits, same as regular scale up. Might be another opportunity to extract some code from existing scale up to a new shared function.

Reply:
Good catch. Will add the resource limit check and extract some common functions in the next iteration.